backend/modules/publishing/service.py

```
@@ -1 +1,603 @@
+                if 'scenarios' in temp_pdf_url:
+                    # Find the scenarios part and extract the key
+                    idx = temp_pdf_url.find('scenarios/')
+                    temp_key = temp_pdf_url[idx:]
+                    pdf_bytes = await s3_service.download_file(temp_key)
+                    if pdf_bytes:
+                        s3_url = await s3_service.upload_from_bytes(pdf_bytes, s3_key, file_type)
+                        if s3_url:
+                            self.repository.create_or_update_scenario_file(
+                                simulation.id, filename, s3_url, file_size, file_type
+                            )
+                            setattr(simulation, 'case_study_url', s3_url)
```

Ruff (0.14.7)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)